python/singa/autograd.py

```diff
@@ -354,6 +354,41 @@ def backward(self, dy):
         return singa.__mul__(dy, dx)
 
 
+def relu(x):
+    return ReLU()(x)[0]
+
+class Mean(Operation):
+    def __init__(self):
+        super(Mean, self).__init__()
+
+    def forward(self, *l):
+        """
+        Args:
+            l: a list of CTensor
+            element-wise mean operator
+        Returns:
+            a new CTensor
+        """
+        if training:
+            self.l = len(l)
+        assert(len(l)>0);
+        x = l[0]
+        for i in range(1,len(l)):
+            x = singa.__add__(x,l[i])
```

Comment:
this is not efficient. you can create an empty tensor and then accumulate each of l into it.

Reply:
use inplace addition.
like x+= l[i]

Reply:
```
x=tensor and set tensor to zero
for i in range(len(l)):
    x += l[i]
```
Is that what you mean?

Reply:
yes, I just changed the code